clusterloader2/pkg/ticker/timer.go

```diff
@@ -17,18 +17,67 @@ limitations under the License.
 package ticker
 
 import (
+	"fmt"
+	"math/rand"
 	"time"
 
 	"k8s.io/perf-tests/clusterloader2/api"
 )
 
-func newTimer(ch chan<- struct{}, ts *api.TuningSet) *time.Timer {
+func newTimer(ch chan<- struct{}, ts *api.TuningSet) (*time.Timer, error) {
 	var t *time.Timer
+	getTicks, err := getTicksFunc(ts)
+	if err != nil {
+		return nil, err
+	}
+	getDelay, err := getDelayFunc(ts)
+	if err != nil {
+		return nil, err
+	}
 	t = time.AfterFunc(ts.InitialDelay, func() {
-		// TODO: Implement other tuning sets.
-		d := 10 * time.Millisecond
-		ch <- struct{}{}
-		t.Reset(d)
+		for i := int32(0); i < getTicks(); i++ {
+			ch <- struct{}{}
+		}
+		t.Reset(getDelay())
 	})
-	return t
+	return t, nil
+}
+
+// getTicksFunc returns function that specifies number of ticks that should be provided after delay.
+func getTicksFunc(ts *api.TuningSet) (func() int32, error) {
+	// for QpsLoad and RandomizedLoad, there is always 1 tick.
+	if ts.QpsLoad != nil || ts.RandomizedLoad != nil {
+		return func() int32 { return int32(1) }, nil
+	}
+	// for SteppedLoad, there is number of ticks equaled to BurstSize.
```

Comment:
Maybe `For SteppedLoad, 'BurstSize' many ticks are created at a time`?

Reply:
done